tests/integration/long/test_ssl.py

```diff
@@ -193,6 +242,62 @@ def test_can_connect_with_ssl_client_auth(self):
 
         cluster.shutdown()
 
+    def test_can_connect_with_ssl_client_auth_host_name(self):
+        """
+        Test to validate that we can connect to a C* cluster that has client_auth enabled, and hostmatching
+
+        This test will setup and use a c* cluster that has client authentication enabled. It will then attempt
+        to connect using valid client keys, and certs (that are in the server's truststore), and attempt to preform some
+        basic operations, with match_hostname specified
+        @jira_ticket PYTHON-296
+        @since 3.3
+
+        @expected_result The client can connect via SSL and preform some basic operations
+
+        @test_category connection:ssl
+        """
+
+        # Need to get absolute paths for certs/key
+        abs_path_ca_cert_path = os.path.abspath(CLIENT_CA_CERTS)
+        abs_driver_keyfile = os.path.abspath(DRIVER_KEYFILE)
+        abs_driver_certfile = os.path.abspath(DRIVER_CERTFILE)
+
+        tries = 0
+        while True:
+            if tries > 5:
+                raise RuntimeError("Failed to connect to SSL cluster after 5 attempts")
+            try:
+                cluster = Cluster(protocol_version=PROTOCOL_VERSION, ssl_options={'ca_certs': abs_path_ca_cert_path,
+                                                                                  'ssl_version': ssl.PROTOCOL_TLSv1,
+                                                                                  'keyfile': abs_driver_keyfile,
```

Comment:
I'm not sure this tests any more than the one above. Because the server is not doing mutual authentication, I think the only thing in play here is the ca cert (same as above).